src/com_tjvendors/site/includes/vendor.php

```diff
@@ -0,0 +1,455 @@
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getLogo()
+	{
+		if (empty($this->vendor_logo))
+		{
+			return Uri::root() . "administrator/components/com_tjvendors/assets/images/default.png";
+		}
+
+		return Uri::root() . $this->vendor_logo;
+	}
+
+	/**
+	 * Check whether the vendor is associated with the provided client
+	 *
+	 * @param   string  $client  The client name to check integration eg. com_jticketing, com_tjllms
+	 *
+	 * @return  bool True on success
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function isAssociatedToClient($client = '')
+	{
+		static $clientExist = null;
+
+		if (!$this->vendor_id)
+		{
+			return false;
+		}
+
+		if ($clientExist[$client] != null)
+		{
+			return $clientExist[$client];
+		}
+
+		/** @var $table TjvendorsTableVendor */
+		$table = TJVendors::table("vendor");
+		$clientExist[$client] = $table->load(array('vendor_id' => $this->vendor_id, 'client' => $client));
+
+		return $clientExist[$client];
+	}
+
+	/**
+	 * Set client properties to the vendor object
+	 *
+	 * @param   string  $client  The client name to check integration eg. com_jticketing, com_tjllms
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function setClient($client)
+	{
+		// Check for the client in the xref
+		$xreftable = TJVendors::table("vendorclientxref");
+
+		if (!$xreftable->load(array('vendor_id' => $this->vendor_id, 'client' => $client)))
+		{
+			return false;
+		}
+
+		$this->client      = $xreftable->get('client');
+		$this->approved    = (int) $xreftable->get('approved');
+
+		// State property override by the client
+		$this->state       = (int) $xreftable->get('state');
+
+		// Expecting RuntimeException thrown by the Joomla\Registry\Format\Json class
+		try
+		{
+			// Params properties are overrided by the client params
+			$clientParams = new Registry($xreftable->get('params'));
+			$vendorParam  = new Registry($this->params);
+			$vendorParam->merge($clientParams);
+			$this->params      = $vendorParam->toString();
+
+			if (!empty($this->params))
+			{
+				$this->payment_gateway = $vendorParam->get('payment_gateway');
+			}
+		}
+		catch (Exception $e)
+		{
+			// Don't throw an error
+		}
+
+		return true;
+	}
+
+	/**
+	 * Method to return all the key and values of all properties
+	 *
+	 * @param   boolean  $public  If true, returns only the public properties.
+	 *
+	 * @return  array
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getProperties($public = true)
+	{
+		return get_object_vars($this);
+	}
+
+	/**
+	 * Method to get gateway details
+	 *
+	 * @return  boolean  True if empty false otherwise
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function checkPaymentGateway()
+	{
+		// Check for the client in the xref
+		$xreftable = TJVendors::table("vendorclientxref");
+
+		if (!$xreftable->load(array('vendor_id' => $this->vendor_id, 'client' => $this->client)))
+		{
+			return false;
+		}
+
+		$params = json_decode($xreftable->params);
```

Comment:
@niteshkesarkar params already merged in the vendor object in the `setClient` method read the details from the object itself.

Reply:
@thite-amol - Used value from `$this->payment_gateway` property.